pytorch/serve
# updated object detector handler to use only default cuda device for torchvision version less then 0.6

## Description

There was bug in torchvision "0.5.0" due to which the object detector models like faster-rcnn, mask-rcnn etc. did not work on multiple GPU devices. Thus the object detector default handler used "cuda:0" device for execution on GPU instances.

With torchvision "0.6.0" release this issue has been fixed. This PR makes the device selection logic conditional for version "0.5.0" and below. Also made changes to use default cuda device instead of "cuda:0"

Also enhanced sanity script to validate all default handlers.

Fixes #104 

## Type of change


## Feature/Issue validation/testing

 - Install torchvision 0.5.0
 - register fastrcnn model with 4 workers [should be successful]
 - run inference 4 times [should be successful]
 - check nvidi-smi output, all workers should be created on default cuda device.

 - upgrade torchvision to 0.6.0
 - register fastrcnn model with 4 workers [should be successful]
 - run inference 4 times [should be successful]
 - check nvidi-smi output, all workers should be created on different cuda device.

Logs :

Install from source : 
[install_from_src_104.txt](https://github.com/pytorch/serve/files/4668710/install_from_src_104.txt)

Test Object detector on multiple GPUs (p3.8xlarge) : 
[test_object_detector_gpu_104.log](https://github.com/pytorch/serve/files/4668714/test_object_detector_gpu_104.log)


## Checklist:
